src/lib/shapelib/ShapeComponent.tsx

```diff
@@ -1,16 +1,68 @@
-import React, { useEffect, useRef } from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import { drawShape, ShapeProps } from './core';
 
-export const ShapeComponent: React.FC<ShapeProps & React.HTMLAttributes<HTMLElement>> = (props) => {
-  const refToDiv = useRef(null);
+export interface ShapeComponentProps extends ShapeProps { 
+  className?: string;
+  style?: React.CSSProperties;
+  debug?: boolean;
+}
+
+const defaultShapeStyle: React.CSSProperties = {
+  position: 'absolute',
+  top: 0,
+  left: 0,
+  width: '100%',
+  height: '100%',
+  zIndex: -1,
+}
+
+const debugShapeStyle: React.CSSProperties = {
+  border: '2px dashed cyan',
+  boxSizing: 'border-box',
+};
+
+export const ShapeComponent: React.FC<ShapeComponentProps> = ({
+  className,
+  style: customShapeStyle,
+  debug,
+  ...shapeProps
+}) => {
+  const [rootNode, setRootNode] = useState<HTMLDivElement | null>(null);
+
+  const onRefChange = useCallback((node: HTMLDivElement) => {
+    setRootNode(node);
+  }, []);
 
   useEffect(() => {
-    if (refToDiv.current) {
-      const ref = refToDiv.current as HTMLElement;
-      ref.textContent = '';
-      drawShape({ element: ref, ...props });
-    }
-  });
```

Comment:
useEffect is called on every re-render if no dependencies are specified.

Reply:
+1 to adding dependencies to useEffect here.
Not sure why the change from useRef to useState + onRefChange?